backend/test_observer/controllers/test_executions/reruns.py

```diff
@@ -18,10 +18,15 @@
 
 
 @router.post("/reruns")
-def create_a_rerun_request(request: RerunRequest, db: Session = Depends(get_db)):
-    te = db.get(TestExecution, request.test_execution_id)
+def create_rerun_requests(request: RerunRequest, db: Session = Depends(get_db)):
+    for test_execution_id in request.test_execution_ids:
+        _create_rerun_request(test_execution_id, db)
+
+
+def _create_rerun_request(test_execution_id: int, db: Session) -> None:
+    te = db.get(TestExecution, test_execution_id)
     if not te:
-        msg = f"No test execution with id {request.test_execution_id} found"
+        msg = f"No test execution with id {test_execution_id} found"
```

Comment:
This way of checking whether the Test Execution ID is valid might lead us to the API behaving somewhat inconsistent. For example, if only one of the test execution IDs provided is invalid, then all test executions appearing before it would have been rerun, while the ones after it wont.

Maybe a better way we can handle this is to firstly filter just test executions whose id is in the set, and then create `TestExecutionRerunRequest` just for those? Or maybe just ignore the ones that have invalid ID, instead of raising the `HTTPException`

Reply:
Good point. I opted to rerun everything that can be rerun. Then return a list of what was rerun. If nothing gets rerun then I return a status code of 404. If everything gets rerun then a status code of 200. And if some rerun and others don't then I return 207.